### PR TITLE
Fetch user initials from UIServer

### PR DESCRIPTION
**Pull request by markgrahamdawson**
Closes https://github.com/cylc/cylc-ui/issues/1504

The userprofile endpoint has been extended in https://github.com/cylc/cylc-uiserver/pull/510 so it now provides user initials.

If for whatever reason the initials field is empty the avatar will be a user symbol
image

And if the initials field is populated the avatar will look like one of the two 
image
image

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by markgrahamdawson** [comment]
Is a changes.md entry required for this one?

File: tests/unit/model/user.model.spec.js [hunk 1]
```diff
@@ -27,14 +27,18 @@ describe('User model', () => {
...
       expect(user.username).to.equal(name)
       expect(user.groups.length === 2).to.equal(admin)
       expect(user.admin).to.equal(true)
       expect(user.created).to.equal(created)
       expect(user.server).to.equal(server)
       expect(user.owner).to.equal(owner)
       expect(user.permissions).to.equal(permissions)
+      expect(user.mode).to.equal(mode)
+      expect(user.initials).to.equal(initials)
     })
```
Comment (markgrahamdawson) [inline tests/unit/model/user.model.spec.js:42]: There are more things coming back from the userprofile endpoint now although they are not actually being used, for example...
avatar_url
color
should these be added to the Vue user model and service and tests as well?

**Comment by oliver-sanders** [comment]
No need for a changelog entry for this ticket (no one is going to check the cylc-ui changelog in the first place to major things only)

File: tests/unit/model/user.model.spec.js (hunk 1 above)
Comment (oliver-sanders) [reply tests/unit/model/user.model.spec.js:42]: Maybe, we can do that later.

Opened an issue for it: https://github.com/cylc/cylc-ui/issues/1519

File: src/components/cylc/workflow/Toolbar.vue [hunk 2]
```diff
@@ -119,7 +119,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
       </v-btn>
       <v-btn icon size="small">
         <v-avatar color="primary" size="small">
-          {{ userInitials }}
+          <v-icon v-if="!user.initials" :icon="$options.icons.mdiAccountEdit"></v-icon>
```
Comment (oliver-sanders) [inline src/components/cylc/workflow/Toolbar.vue:122]: I'm not sure how initials could end up undefined, so I don't think it's strictly necessary to handle this case, but there's no harm in it.

**Review by oliver-sanders — APPROVED** [review]
Looks good.

Note to reviewers, PAM doesn't return a helpful initials field, but other authenticators can.

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**